ccloud client issue `internal/provider/ccloud/ccloud_client.go:300`

Hi, not sure if this provider still supposed to work, but lets try to ask...

My terraform resource definition example:
```hcl
resource "cockroach-extra_cluster_setting" "this" {

  cluster_id    = "cluster_id"
  setting_name  = "sql.log.slow_query.latency_threshold"
  setting_value = "500ms"
}
```

I am getting the following error:
```
│ Error: Plugin did not respond
│
│ The plugin encountered an error, and failed to respond to the plugin6.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain more details.
╵

Stack trace from the terraform-provider-cockroach-extra_v0.0.18 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xb1fc73]

goroutine 32 [running]:
github.com/nrfcloud/terraform-provider-cockroach-extra/internal/provider/ccloud.(*CcloudClient).getOrCreateConPool(0xc000734330, {0x16b3120, 0xc0006597a0}, {0xc0001326f0, 0x24}, 0x1?, {0x12873d3, 0x9})
	github.com/nrfcloud/terraform-provider-cockroach-extra/internal/provider/ccloud/ccloud_client.go:300 +0x213
github.com/nrfcloud/terraform-provider-cockroach-extra/internal/provider/ccloud.(*CcloudClient).updateUserExpiration(0x1f93c60?, {0x16b3120?, 0xc0006597a0?}, {0xc0001326f0?, 0x101f2a0?}, 0xc00007e3a0)
	github.com/nrfcloud/terraform-provider-cockroach-extra/internal/provider/ccloud/ccloud_client.go:167 +0x4c
github.com/nrfcloud/terraform-provider-cockroach-extra/internal/provider/ccloud.(*CcloudClient).createTempUser(0xc000734330, {0x16b3120, 0xc0006597a0}, {0xc0001326f0, 0x24})
	github.com/nrfcloud/terraform-provider-cockroach-extra/internal/provider/ccloud/ccloud_client.go:150 +0x646
github.com/nrfcloud/terraform-provider-cockroach-extra/internal/provider/ccloud.(*CcloudClient).getOrCreateTempUser(0x1fc38a0?, {0x16b3120, 0xc0006597a0}, 0x1fc38a0?, {0xc0001326f0, 0x24})
	github.com/nrfcloud/terraform-provider-cockroach-extra/internal/provider/ccloud/ccloud_client.go:181 +0x15b
github.com/nrfcloud/terraform-provider-cockroach-extra/internal/provider/ccloud.SqlConWithTempUser[...]({0x16b3120, 0xc0006597a0}, 0x0, {0xc0001326f0?, 0x24}, {0x12873d3, 0x9}, 0xc000670df8)
	github.com/nrfcloud/terraform-provider-cockroach-extra/internal/provider/ccloud/ccloud_client.go:334 +0x129
github.com/nrfcloud/terraform-provider-cockroach-extra/internal/provider/resources.(*ClusterSettingResource).setClusterSetting(0x16b3120?, {0x16b3120?, 0xc0006597a0?}, {0xc0001326f0?, 0x0?}, {0xc000132720?, 0x0?}, {0xc0000121aa?, 0x16b3120?})
	github.com/nrfcloud/terraform-provider-cockroach-extra/internal/provider/resources/cluster_setting_resource.go:89 +0x96
github.com/nrfcloud/terraform-provider-cockroach-extra/internal/provider/resources.(*ClusterSettingResource).Create(0xd?, {0x16b3120, 0xc0006597a0}, {{{{0x16b8198, 0xc000710510}, {0x109f480, 0xc000710450}}, {0x16b9ed8, 0xc000101cc0}}, {{{0x16b8198, ...}, ...}, ...}, ...}, ...)
	github.com/nrfcloud/terraform-provider-cockroach-extra/internal/provider/resources/cluster_setting_resource.go:107 +0x2d2
github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(*Server).CreateResource(0xc00018bce0, {0x16b3120, 0xc0006597a0}, 0xc000671390, 0xc000671330)
	github.com/hashicorp/terraform-plugin-framework@v1.4.2/internal/fwserver/server_createresource.go:101 +0x578
github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(*Server).ApplyResourceChange(0xc0006714e8?, {0x16b3120, 0xc0006597a0}, 0xc0000a8a50, 0xc0006714e8)
	github.com/hashicorp/terraform-plugin-framework@v1.4.2/internal/fwserver/server_applyresourcechange.go:57 +0x4a5
github.com/hashicorp/terraform-plugin-framework/internal/proto6server.(*Server).ApplyResourceChange(0xc00018bce0, {0x16b3120?, 0xc0006594d0?}, 0xc0000a89b0)
	github.com/hashicorp/terraform-plugin-framework@v1.4.2/internal/proto6server/server_applyresourcechange.go:55 +0x3ff
github.com/hashicorp/terraform-plugin-go/tfprotov6/tf6server.(*server).ApplyResourceChange(0xc0005a3ea0, {0x16b3120?, 0xc000658840?}, 0xc00041c2a0)
	github.com/hashicorp/terraform-plugin-go@v0.19.0/tfprotov6/tf6server/server.go:857 +0x56a
github.com/hashicorp/terraform-plugin-go/tfprotov6/internal/tfplugin6._Provider_ApplyResourceChange_Handler({0x1235020?, 0xc0005a3ea0}, {0x16b3120, 0xc000658840}, 0xc00041c230, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.19.0/tfprotov6/internal/tfplugin6/tfplugin6_grpc.pb.go:467 +0x169
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0001203c0, {0x16b8d88, 0xc0005d8680}, 0xc0006d0fc0, 0xc000658a50, 0x1f84578, 0x0)
	google.golang.org/grpc@v1.58.3/server.go:1374 +0xde7
google.golang.org/grpc.(*Server).handleStream(0xc0001203c0, {0x16b8d88, 0xc0005d8680}, 0xc0006d0fc0, 0x0)
	google.golang.org/grpc@v1.58.3/server.go:1751 +0x9e7
google.golang.org/grpc.(*Server).serveStreams.func1.1()
	google.golang.org/grpc@v1.58.3/server.go:986 +0xbb
created by google.golang.org/grpc.(*Server).serveStreams.func1 in goroutine 26
	google.golang.org/grpc@v1.58.3/server.go:997 +0x145

Error: The terraform-provider-cockroach-extra_v0.0.18 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

Am i doing something wrong?

--- a/internal/provider/ccloud/ccloud_client.go
+++ b/internal/provider/ccloud/ccloud_client.go
@@ -253,7 +253,8 @@ func (c *CcloudClient) getConnectionOptions(ctx context.Context, clusterId strin
 		return nil, err
 	}
 
-	if processCloudResponse(resp, nil) != nil {
+	err = processCloudResponse(resp, nil)
+	if err != nil {
 		return nil, err
 	}
 

